### PR TITLE
Improve ways to create objects

### PR DESCRIPTION
**Pull request by apastan**
### Description

Edit the alternative ways of creating new objects.

### CODE REVIEW

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
I don't get this change. What's the motivation? (BTW, "constructor function" is not a code entity so it should not be code-formatted, and you can't invoke a class.)

**Comment by apastan** [comment]
Hi. If I understand the context correctly in the second sentence of the paragraph it is talked about the ways of creating objects.

There are 3 way to create an object:
- Object initializer syntax
- by invoking function as constructor (or via Class) with `new` operator
- method Object.create()

The same idea is described in article: [Creating new objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects#creating_new_objects)
- [Using object initializers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects#using_object_initializers)
- [Using a constructor function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects#using_a_constructor_function)
- [Using the Object.create() method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects#using_the_object.create_method)

As I understand these 3 methods were specified for the following reasons:
- Object initializer. Most simple way to create object with prototype `Object.prototype`. It also allow to use
- Object.create(). It is useful because Object initializer doesn’t allow specify prototype of newly created object and specify property descriptors.
- by invoking function as constructor (or via Class) with `new` operator. We should define the needed object type by writing a custom constructor function (specify function name, properties and methods for every instance) or using build-in constructor function (**Object**, Array, Function, Number, String, Boolean, etc.). And then we can create new object by invoking function as constructor. 

In my opinion, `new Object` doesn't provide any special behavior and can be used in 2 cases:
- creating empty object with prototype `Object.prototype`. But it's not very useful because we can't specify prototype for newly created object, or object properties, or use syntax of Object initializer). We can get the same result simply using `{}`.
- for type coercion: coerce a primitive value to object. But it is necessary to handle this operation with caution. 

If we provide as argument `null`, `undefined` or invoke `Object` without arguments Object constructor will create an empty object with prototype `Object.prototype`:

```js
new Object(); // {}
Object(); // {}
new Object(undefined); // {}
Object(undefined); // {}
new Object(null); // {}
Object(null); // {}

Object.getPrototypeOf( new Object(null) ) === Object.prototype; // true
Object.getPrototypeOf( Object(null) )  === Object.prototype; // true

Object.getPrototypeOf( new Object(undefined) ) === Object.prototype; // true
Object.getPrototypeOf( Object(undefined) ) === Object.prototype; // true

Object.getPrototypeOf( new Object() ) === Object.prototype; // true
Object.getPrototypeOf( Object() ) === Object.prototype; // true
```

If we provide as argument a primitive `Object` will performs a type conversion to object:

```js
new Object(5); // Number {5}
new Object(true); // Boolean {true}
new Object("Hi"); // String {'Hi'}
new Object(1n); // BigInt {1n}

Object.getPrototypeOf( new Object(5) ) === Number.prototype; // true
Object.getPrototypeOf( Object(5) ) === Number.prototype; // true
Object.getPrototypeOf( new Number(5) ) === Number.prototype; // true

Object.getPrototypeOf( new Object(true) ) === Boolean.prototype; // true
Object.getPrototypeOf( Object(true) ) === Boolean.prototype; // true
Object.getPrototypeOf( new Boolean(true) ) === Boolean.prototype; // true

Object.getPrototypeOf( new Object("Hi") ) === String.prototype; // true
Object.getPrototypeOf( Object("Hi") ) === String.prototype; // true
Object.getPrototypeOf( new String("Hi") ) === String.prototype; // true

Object.getPrototypeOf( new Object("Hi") ) === String.prototype; // true
Object.getPrototypeOf( Object("Hi") ) === String.prototype; // true
Object.getPrototypeOf( new String("Hi") ) === String.prototype; // true

Object.getPrototypeOf( new Object(1n) ) === String.prototype; // true
Object.getPrototypeOf( Object(1n) ) === String.prototype; // true
```

If we provide as argument an object (not primitive value) `Object` will return this object without creating new object:

```js
const obj = {x: 0, y: 0};
const f = () => {};

new Object(obj) === obj; //true
Object(obj) === obj; //true

new Object(f) === f; //true
Object(f) === f; //true
```

File: files/en-us/web/javascript/reference/operators/object_initializer/index.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ browser-compat: javascript.operators.object_initializer
 
 {{JsSidebar("Operators")}}
 
-An **object initializer** is a comma-delimited list of zero or more pairs of property names and associated values of an object, enclosed in curly braces (`{}`). Objects can also be initialized using [`new Object()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/Object) or [`Object.create()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create).
+An **object initializer** is a comma-delimited list of zero or more pairs of property names and associated values of an object, enclosed in curly braces (`{}`). Objects can also be initialized using [`Object.create()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create) or [by invoking function as a constructor](/en-US/docs/Web/JavaScript/Guide/Working_with_Objects#using_a_constructor_function) (or via [`class`](/en-US/docs/Web/JavaScript/Reference/Classes) syntax) with the [`new`](/en-US/docs/Web/JavaScript/Reference/Operators/new) operator.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/operators/object_initializer/index.md:10]:
```suggestion
An **object initializer** is a comma-delimited list of zero or more pairs of property names and associated values of an object, enclosed in curly braces (`{}`). Objects can also be initialized using [`Object.create()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create) or [by invoking a constructor function](/en-US/docs/Web/JavaScript/Guide/Working_with_Objects#using_a_constructor_function) with the [`new`](/en-US/docs/Web/JavaScript/Reference/Operators/new) operator.
```

**Review by Josh-Cena — APPROVED** [review]
OK, I get what you mean. I agree this looks more sensible and encompasses the same information as before.

**Comment by Josh-Cena** [comment]
Thank you for following up on this PR!

**Pull request merged**